src/components/Accounts/Actions/ActionModal.js

```diff
@@ -0,0 +1,509 @@
+      notify
+    } = this.state;
+
+    const selected = calculateSelectedAmount(accounts);
+    const remaining = parseFloat(balance - amount).toFixed(2);
+    const staffPlaceholder = formatMessage({
+      id: 'ui-users.accounts.placeholder.additional'
+    },
+    {
+      action: formatMessage({ id: `ui-users.accounts.actions.${action}` }),
+      type: formatMessage({ id: `ui-users.accounts.${(commentRequired) ? 'required' : 'optional'}` })
+    });
+
+    const ownerOptions = owners.filter(o => o.owner !== 'Shared').map(o => ({ value: o.id, label: o.owner }));
+
+    let options = (action === 'payment') ? data.filter(d => (d.ownerId === (accounts.length > 1 ? ownerId : (accounts[0] || {}).ownerId))) : data;
+    options = _.uniqBy(options.map(o => ({ id: o.id, label: o[label] })), 'label');
+
+    return (
+      <Modal
+        id={`${action}-modal`}
+        open={open}
+        label={<FormattedMessage id={`ui-users.accounts.${action}.modalLabel`} />}
+        onClose={this.onClose}
+        size="medium"
+        dismissible
+      >
+        <form>
+          <Row>
+            <Col xs>{this.renderModalLabel()}</Col>
+          </Row>
+          <br />
+          <Row>
+            <Col xs={5}>
+              <Row end="xs">
+                <Col xs={7}>
+                  <FormattedMessage id="ui-users.accounts.totalOwed" />
+                  {':'}
+                </Col>
+                <Col xs={4}>
+                  {balance}
+                </Col>
+              </Row>
+              <Row end="xs">
+                <Col xs={7}>
+                  <FormattedMessage id="ui-users.accounts.selectedAmount" />
+                  {':'}
+                </Col>
+                <Col xs={4}>
+                  {selected}
+                </Col>
+              </Row>
+              <Row end="xs">
+                <Col xs={7}>
+                  <b>
+                    <FormattedMessage id={`ui-users.accounts.${action}.amount`} />
+                    {'*:'}
+                  </b>
+                </Col>
+                <Col xs={4} className={css.customCol}>
+                  <div>
+                    <Field
+                      id="amount"
+                      name="amount"
+                      component={TextField}
+                      onChange={this.onChangeAmount}
+                      hasClearIcon={false}
+                      fullWidth
+                      marginBottom0
+                      autoFocus
+                      required
+                    />
+                  </div>
+                </Col>
+              </Row>
+              <Row end="xs">
+                <Col xs={7}>
+                  <FormattedMessage id="ui-users.accounts.remainingAmount" />
+                  {':'}
+                </Col>
+                <Col xs={4}>
+                  {remaining}
+                </Col>
+              </Row>
+            </Col>
+            {(action === 'payment' && accounts.length > 1) &&
```

Comment:
Move this check to a method `isPaymentAction(action)` and use it all over the file.